### PR TITLE
fw_iptables.c: use libiptc instead of command iptables.

### PR DESCRIPTION
**Pull request by gigibox**
libiptc is the library that is used to communicate with netfilter, the
internal kernel code in charge of firewalling and packet filtering. This
code and iptables were written by Paul "Rusty" Russell. iptables was
developed using libiptc calls to get the job done.

Signed-off-by: ZengFei Zhang <zhangzengfei@kunteng.org>

### CODE REVIEW

File: src/fw3_iptc.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+void
+fw3_ipt_close(struct fw3_ipt_handle *h);
+
+int
+fw3_ipt_commit(struct fw3_ipt_handle *h);
+
+int
+fw3_ipt_rule_append(struct fw3_ipt_handle *handle, char *command);
+
+#define LOCK_FW3_IPT() do { \
```
Comment (liudf0716) [inline src/fw3_iptc.h]: I think no need to introduce new lock, u can use config lock instead.

File: src/xtables-10.h [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+static inline void
+fw3_xt_print_matches(void *ip, struct xtables_rule_match *matches)
+{
+	struct xtables_rule_match *rm;
+	struct xtables_match *m;
+
+	for (rm = matches; rm; rm = rm->next)
+	{
+		m = rm->match;
+		printf(" -m %s", fw3_xt_get_match_name(m));
```
Comment (liudf0716) [inline src/xtables-10.h]: please use debug instead of printf.

File: src/xtables-5.h [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	g->opts = xtables_merge_options(g->opts, t->extra_opts, &t->option_offset);
+}
+
+static inline void
+fw3_xt_print_matches(void *ip, struct xtables_rule_match *matches)
+{
+	struct xtables_rule_match *rm;
+	struct xtables_match *m;
+
+	printf(" ");
```
Comment (liudf0716) [inline src/xtables-5.h]: replace with debug

Comment (gigibox) [inline src/xtables-5.h]: ok! do it now.

**Comment by gigibox** [comment]
@liudf0716 All work is done.

**Pull request merged**